digdag-core/src/main/java/io/digdag/core/ErrorReporter.java

```diff
@@ -0,0 +1,16 @@
+package io.digdag.core;
+
+public interface ErrorReporter
```

Comment:
perhaps naming is not good but, that's because threads retry operations when uncaught exceptions happen. For example, `WorkflowExecutorLoop.run` may catch uncaught error when database transaction caused error (such as DNS error, statement timeout, etc.). Appropriate recovery behavior is another discussion but retrying the hole operation is an option. Retrying won't be an option with `Thread.UncaughtExceptionHandler` handler.